public/main.js

```diff
@@ -162,29 +162,43 @@ let getExtraInput = (event) => {
 
 let displayRecipes = (recipes) => {
     recipes.forEach(item => {
-        console.log(recipes);
         let recipesContainer = document.querySelector('.recipes-container');
+        let recipeHeart = document.createElement('h1');
         let recipe = document.createElement('div');
         let recipeName = document.createElement('p');
         let recipeURL = document.createElement('a');
         let recipePhoto = document.createElement('img');
         recipe.classList.add('recipe');
+        recipeHeart.textContent= "♥";
         recipeName.classList.add('recipe-name');
         recipePhoto.classList.add('recipe-photo');
         recipeURL.classList.add('recipe-url');
-        console.log(item);
         recipeName.textContent = item[0];
         recipePhoto.setAttribute('src', item[2]);
+        recipeHeart.setAttribute('class', 'heart')
         recipeURL.setAttribute('href', item[1]);
         recipeURL.setAttribute('target', '_blank');
         recipeURL.setAttribute('rel', 'noopener noreferrer');
         recipeURL.appendChild(recipeName);
         recipeURL.appendChild(recipePhoto);
         recipe.appendChild(recipeURL);
+        recipe.appendChild(recipeHeart);
+        recipeHeart.addEventListener('click',() => likeRecipe(item[4]));
         recipesContainer.appendChild(recipe);
     });
 };
 
+let likeRecipe = (id) => {
+    event.target.classList.toggle("selected");
+    return fetch('/like', {
+        method: "POST",
+        body: JSON.stringify(id),
+        headers: {'Content-Type': 'application/json', 
+        'authorization': JSON.parse(localStorage.getItem("token"))}
+    }).then(res => res.json())
+    .then(res => console.log(res))
```

Comment:
I wonder if we should refactor at one point to have a function that grabs the token, since we know do it in so many places.

Reply:
that wouldn't be a bad thing! i agree - it is all over the place